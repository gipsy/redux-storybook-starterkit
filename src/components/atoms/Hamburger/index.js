import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { palette } from 'styled-theme'
import { ifProp } from 'styled-tools'
import { placeholder, rem, transitions } from 'polished'

const Wrapper = styled.span`
  display: inline-block;
  color: ${ifProp('palette', palette({ grayscale: 0 }, 1), 'currentcolor')};
  margin: 0.1rem;
  box-sizing: border-box;
  cursor: pointer;
`

const HamburgerSpan = styled.span`
  display: block;
  width: 20px;
  height: 3px;
  margin-bottom: 3px;
  position: relative;
  background-color: ${ifProp('palette', palette({ grayscale: 0 }, 1), 'currentcolor')};
  z-index: 1;
  transform-origin: 4px 0px;
  ${transitions('0.5s cubic-bezier(0.77,0.2,0.05,1.0),background 0.5s cubic-bezier(0.77,0.2,0.05,1.0),opacity 0.55s ease')};

  &:first-child {
    transform-origin: 0% 0%;
  }

  &:nth-last-child(2) {
    transform-origin: 0% 100%;
  }
`

const Hamburger = ({ ...props }) => {
  return (
    <Wrapper {...props}>
      <HamburgerSpan />
      <HamburgerSpan />
      <HamburgerSpan />
    </Wrapper>
  )
}

Hamburger.propTypes = {
  palette: PropTypes.string,
  reverse: PropTypes.bool
}

export default Hamburger
