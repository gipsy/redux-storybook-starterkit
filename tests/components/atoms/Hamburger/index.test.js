import React from 'react'
import { shallow } from 'enzyme'
import Hamburger from '~/src/components/atoms/Hamburger'

const wrap = (props = {}) => shallow(<Hamburger {...props} />).dive()

test('<Hamburger /> renders props', () => {
  const wrapper = wrap({ reverse: true, palette: 'primary' })
  expect(wrapper).toMatchSnapshot()
})
