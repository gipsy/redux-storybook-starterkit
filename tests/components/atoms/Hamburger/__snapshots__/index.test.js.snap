// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> renders props 1`] = `
<span
  className="s185djk0-0 cVrnen"
>
  <styled.span />
  <styled.span />
  <styled.span />
</span>
`;
